src/software/thunderscope/field/path_layer.py

```diff
@@ -31,18 +29,68 @@ def paint(self, painter, option, widget):
         :param widget: The widget that we are painting on
 
         """
-        paths = self.path_visualization_buffer.get(block=False)
+        primitive_set = self.primitive_set_buffer.get(
+            block=False
+        ).robot_primitives.values()
+        paths = [
+            primitive.move.motion_control.path
+            for primitive in primitive_set
+            if primitive.HasField("move")
+        ]
+
+        requested_destinations = [
+            primitive.move.motion_control.requested_destination
+            for primitive in primitive_set
+            if primitive.HasField("move")
+        ]
+
         painter.setPen(
             pg.mkPen(Colors.NAVIGATOR_PATH_COLOR, width=constants.LINE_WIDTH)
         )
 
-        for path in paths.path:
+        for path in paths:
             polygon_points = [
                 QtCore.QPoint(
                     int(MILLIMETERS_PER_METER * point.x_meters),
                     int(MILLIMETERS_PER_METER * point.y_meters),
                 )
-                for point in path.point
+                for point in path.points
             ]
             poly = QtGui.QPolygon(polygon_points)
             painter.drawPolyline(poly)
+
+        offset_1 = 20
+        offset_2 = 40
+        for dest in requested_destinations:
+            x_mm = int(MILLIMETERS_PER_METER * dest.x_meters)
+            y_mm = int(MILLIMETERS_PER_METER * dest.y_meters)
+            polygon_points = [
```

Comment:
maybe add a comment here that you're adding a X

Reply:
fixed